### PR TITLE
Line Diagram | Collapsible branches

### PR DESCRIPTION
**Pull request by thecristen**
#### Summary of changes
**Asana Ticket:** [Line Diagram | Collapsible branches](https://app.asana.com/0/385363666817452/1141229989083711/f)

* Adds styles for collapsed and expanded branches. It's a delightful mix of JavaScript logic, SVG attributes, rather complex CSS rules...
* Breaks up the larger React component into a few smaller ones
* Reuses the existing `<ExpandableBlock />` component to handle collapsing/expanding sections of the line diagram
* Bugfix - removes the "View schedule" link from the terminus stops
* IE fix - removes the usage of "unset" in our CSS
* Apparently I'm the first to use a lodash function here... so I moved the package from `devDependencies`.

### Screenshots!

<details>
<summary>IE11 preview</summary>

**Video**: [https://app.crossbrowsertesting.com/public/i31732a965037ca8/livetests/35727589/videos/z9d8a4cb074e2a020941?test_hash=e0aed05e](https://app.crossbrowsertesting.com/public/i31732a965037ca8/livetests/35727589/videos/z9d8a4cb074e2a020941?test_hash=e0aed05e)

</details>

<details>
<summary>Firefox preview</summary>

Screen Shot 2020-01-13 at 14 55 38

</details>

<details>
<summary>Safari preview</summary>

Screen Shot 2020-01-13 at 14 56 50

</details>

<details>
<summary>Chrome preview</summary>

localhost_4001_schedules_Green_line_direction_id=0 (1)

</details>

### CODE REVIEW

**Comment by ingridpierre** [comment]
Really small presentational changes. Can we:
- increase the border radius, so that the expand/collapse control has more rounded shoulders.
- vertically center the three dots more inside the expansion control
- make the lines thinner (it's taking a lot of room in mobile especially), instead of 10px and 16px wide, let's try 8 and 12?

**Current:**
image

**Desired (roughly):**
image

File: apps/site/assets/css/_shuttles-page.scss [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
   @include media-breakpoint-only(xs) {
     &:not(:first-child):not(:last-child) {
-      border-radius: unset;
+      border-radius: inherit;
```
[the remark this answers is not in the document]
Comment (thecristen) [reply apps/site/assets/css/_shuttles-page.scss]: whoops, should have noted that in my summary. one of the broken IE things was that `unset` doesn't work, which I chose to resolve by removing all instances of `unset` from our CSS (turns out this one, that I'd added recently, was the only other one 😅 )

[the remark this answers is not in the document]
Comment (digitalcora) [reply apps/site/assets/css/_shuttles-page.scss]: Nice; let's include that in the final commit message, or else split it into a separate bugfix commit.

File: apps/site/assets/ts/schedule/__tests__/LineDiagramTest.tsx [hunk 2]
```diff
@@ -191,7 +191,9 @@ describe("LineDiagram with branches going outward", () => {
   });
 
   it("identifies stops rather than stations", () => {
-    expect(wrapper.find("h3").text()).toEqual("Stops");
+    expect(wrapper.find(".m-schedule-diagram__heading").text()).toEqual(
```
[the remark this answers is not in the document]
Comment (thecristen) [reply apps/site/assets/ts/schedule/__tests__/LineDiagramTest.tsx]: Using `<ExpandableBlock />` introduced additional h3 elements so I needed to be more specific! I can update this to `h3.m-schedule-diagram__heading` though

[the remark this answers is not in the document]
Comment (digitalcora) [reply apps/site/assets/ts/schedule/__tests__/LineDiagramTest.tsx]: Hm, it's unfortunate that the header level is hard-coded in `ExpandableBlock` — hierarchically we'd want those to be "below" the main diagram heading, not at the same level. Is it possible the diagram header could be an h2, or do we already have h2's on the page?

File: apps/site/assets/ts/schedule/components/line-diagram/ExpandableBranch.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        {branchData.map((branchStop, idx) => (
+          <SingleStop
+            key={branchStop.route_stop.id}
+            stop={branchStop}
+            isLast={idx === branchData.length - 1}
+            onClick={onStopClick(branchStop.route_stop)}
+            color={color}
+          />
+        ))}
+      </>
```
[the remark this answers is not in the document]
Comment (thecristen) [reply apps/site/assets/ts/schedule/components/line-diagram/ExpandableBranch.tsx]: This was because the [ExpandableBlock children prop](https://github.com/mbta/dotcom/blob/master/apps/site/assets/ts/components/ExpandableBlock.tsx#L22) is typed to only expect one ReactElement rather than a list of 'em! 🤷🏻‍♀️

[the remark this answers is not in the document]
Comment (digitalcora) [reply apps/site/assets/ts/schedule/components/line-diagram/ExpandableBranch.tsx]: Ah, it seems this is a built-in React thing and so we can't control the name (or, I guess, whether it's able to accept a list of components).

File: apps/site/assets/ts/schedule/__tests__/LineDiagramTest.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (thecristen) [reply apps/site/assets/ts/schedule/__tests__/LineDiagramTest.tsx]: The "Info & Maps" is an h2 and the route name above the tabs is an h1, so yeah, it'd be a bit of a rabbit hole to adjust the tree hierarchy here dynamically.. might be something to look at down the road.

[the remark this answers is not in the document]
Comment (digitalcora) [reply apps/site/assets/ts/schedule/__tests__/LineDiagramTest.tsx]: I was thinking probably the expando-block component should let you specify an element for the header, defaulting to (say) a `div` so it doesn't mess with any semantic hierarchy you might already have on the page. But as far as tech debt goes, we have vastly more impactful things to tackle 😅

File: apps/site/assets/css/_schedule-page-line-diagram.scss [hunk 4]
```diff
@@ -67,118 +106,230 @@ $line-width: 10px;
...
+
     height: $base-spacing * 2;
+  }
 
-    circle {
-      transform: translateY(35px) scale(1.15);
-    }
+  // hide "View schedules" for final destination(s)
+  .m-schedule-diagram__stop--terminus:not(:first-child) .m-schedule-diagram__footer {
+    display: none;
```
[the remark this answers is not in the document]
Comment (thecristen) [reply apps/site/assets/css/_schedule-page-line-diagram.scss]: Not sure about practical, but certainly possible... resolved in latest commit

File: apps/site/assets/package.json [hunk 5]
```diff
@@ -17,6 +17,7 @@
     "jquery": "> 3.0.0",
     "leaflet": "^1.4.0",
     "leaflet-rotatedmarker": "^0.2.0",
+    "lodash": "^4.17.11",
```
[the remark this answers is not in the document]
Comment (thecristen) [reply apps/site/assets/package.json]: I'm not exactly sure how package-lock is _supposed_ to reflect this change, but I've run the update and committed what it gave me.

**Comment by thecristen** [comment]
@ingridpierre @digitalcora Made some updates that should address the comments here.

**Comment by ingridpierre** [comment]
Looks so good, one teeny adjustment -- can we reduce the padding on .m-schedule-diagram__stop from 1rem all the way around to: 1rem .5rem? Decreasing the horizontal padding wins just a tiny bit more line length, which I'd like to take where I can get it :)

File: apps/site/assets/package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (digitalcora) [reply apps/site/assets/package.json]: I had thought since we moved a package from `devDependencies` to `dependencies` that this would be reflected in the lockfile by that dependency flipping from `"dev": true` to `"dev": false`. But since nothing changed (aside from the usual [optional fights](https://github.com/dependabot/feedback/issues/197)), I'm thinking we already had a package in our production `dependencies` that was pulling in Lodash as a dependency of its own, hence it was already marked as `"dev": false`.

tl;dr: we don't have to commit this change to the package-lock 😅

**Comment by thecristen** [comment]
@ingridpierre done!
@digitalcora I've gone ahead and renamed first/last to origin/destination and also was able to deprecate the `isEnd` prop!

File: apps/site/assets/css/_schedule-page-line-diagram.scss [hunk 6]
```diff
@@ -67,118 +106,234 @@ $line-width: 10px;
...
   //  graphics for the line diagram, and uses negative
   //  margins to counter the padding from its parent
   //  .m-schedule-diagram__stop.
   //  A .m-schedule-diagram__line is drawn for each
-  //  branch, and creates the line. Stop circles are SVGs.
+  //  branch, and creates the line.
   .m-schedule-diagram__lines {
     display: flex;
     margin: calc(-#{$base-spacing} - #{$border-width}*2) 0;
     margin-right: $base-spacing;
```
Comment (ingridpierre) [inline apps/site/assets/css/_schedule-page-line-diagram.scss]: Looks like we also need this right margin to shrink to .5rem also to match the previous change: 

**CURRENT**
image
**DESIRED**
image

**Comment by digitalcora** [comment]
@thecristen Looks like some things you extracted to separate commits are also still present in the commit message for the "main" commit.
